gluten-core/src/main/scala/io/glutenproject/expression/ArithmeticTransformer.scala

```diff
@@ -284,4 +313,14 @@ object BinaryArithmeticTransformer {
         throw new UnsupportedOperationException(s"not currently supported: $other.")
     }
   }
+
+  def createRemainder(left: Expression, right: Expression,
+                   failOnError: Boolean, original: Expression): Expression = {
+    original match {
+      case r: Remainder =>
+        new RemainderTransformer(left, right, failOnError, r)
+      case other =>
+        throw new UnsupportedOperationException(s"not currently supported: $other.")
```

Comment:
It seems using `UnsupportedOperationException` is not proper. `UnsupportedOperationException` is helping us let some unsupported operator/expression correctly fall back through catch statement. Here, I think it is developer's mistake in calling `createRemainder` for other kinds of expressions. Maybe, we can use `RuntimeException` or some other exception here. Right?

Reply:
RemainderTransformer is removed now.